src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -246,11 +246,14 @@ function VisEditor($scope, $route, timefilter, AppState, $window, kbnUrl, courie
 
     // update the searchSource when filters update
     $scope.$listen(queryFilter, 'update', function () {
+      savedVis.searchSource.set('filter', $scope.state.filters);
       $state.save();
+      $scope.vis.forceReload();
     });
 
     // update the searchSource when query updates
     $scope.fetch = function () {
+      savedVis.searchSource.set('query', $scope.state.query);
```

Comment:
Why can we set `query` on every fetch, while we set filters only when they refresh. Don't we have any on query update mechanism that we could listen on?